webapi/_test_data/all_data/permissions-data.ttl

```diff
@@ -316,6 +316,47 @@
                       knora-base:hasPermissions "CR knora-base:Creator|M knora-base:ProjectMember|V knora-base:KnownUser,knora-base:UnknownUser"^^xsd:string .
 
 
+##########################################################
+#
+# WEBERN Project Permissions
+#
+##########################################################
+
+### Administrative Permissions on ProjectMember
+<http://rdfh.ch/permissions/008-a1>
```

Comment:
I think it would also make sense to add the `projectShortcode` to the permission IRI: `http://rdfh.ch/permissions/08AE/008-a1`.

Reply:
Sorry for disrupting, but we feel concerned by your comment @subotic . So, do we say, as a general rule that we should have the `projectShortcode` in every permission? Can we begin to refactor our data accordingly?

Reply:
I can see your point, but I wonder if this is really necessary when you get the project from `knora-base:forProject <http://rdfh.ch/projects/xyz>` in every permission statement. 

Adding a shortcode to the permission IRI here would also mean that a single permission statement only applies to one single project. Is this intended? Could you imagine to have a permission applying to more than one project (multi-user?)? I see, in the case of different project member groups it will be highly recommended to have seperate permission statements for every group. 
But when you look at the statements for the project admins, for instance, they are quite the same for every project:
```
### Administrative Permissions on ProjectAdmin
<http://rdfh.ch/permissions/00XX-aY>

                      rdf:type knora-base:AdministrativePermission ;

                      knora-base:forProject <http://rdfh.ch/projects/xyz> ;

                      knora-base:forGroup knora-base:ProjectAdmin ;

                      knora-base:hasPermissions "ProjectResourceCreateAllPermission|ProjectAdminAllPermission"^^xsd:string .

```

Reply:
I would like to add the `projectShortcode` to the permission IRIs so that all IRIs follow the same rule. At the moment, the permissions are the only ones left without the shortcode in the IRI. We have the project shortcode as part of the IRI in the project, group, resources, and lists.

Permissions will always be only for one project. Otherwise, it would become too complicated.

I don't like exceptions, as they tend to bite one in the ass sooner or later. Also, when writing permissions by hand, it is easier to make sure that the IRI is unique since you only need to make sure that the last part is unique for the project.

The `knora-base:ProjectMember` and `knora-base:ProjectAdmin` groups are exceptions to the rule, which I would like to get rid of in #679.

Reply:
I would like to add the `projectShortcode` to the permission IRIs so that all IRIs follow the same rule. At the moment, the permissions are the only ones left without the shortcode in the IRI. We have the project shortcode as part of the IRI in the project, group, resources, and lists.

Permissions will always be only for one project. Otherwise, it would become too complicated.

I don't like exceptions, as they tend to bite one in the ass sooner or later. Also, when writing permissions by hand, it is easier to make sure that the IRI is unique since you only need to make sure that the last part is unique for the project.

The `knora-base:ProjectMember` and `knora-base:ProjectAdmin` groups are exceptions to the rule, which I would like to get rid of in #679.

Reply:
@mrivoal yes, please :-)

Reply:
@subotic :
> We have the project shortcode as part of the IRI in the project, group, resources, and lists.

For groups too?

Reply:
yes. I thought I mentioned it earlier because the drawings-gods project test data has already the project shortcode as part of the group IRIs.

I'm updating the documentation right now.

Reply:
Ok, I've added the shortcode to the permission IRI's